README.md

```diff
@@ -1,94 +1,250 @@
-- [`./contracts/bindings`](./contracts/bindings) - bindings generated from ABIs - to generate new bindings see [Generate bindings](#generate-bindings) section.
-- [`./install`](./install/) - collection of things to support the various installation, signing and notarization requirements.
-- [`./docs`](./docs) Some small content to be inserted into mac pgk file
+```
+tools-lukso-cli
+│
+└───abis                  // Binary Interfaces of LUKSO Smart Contracts
+│
+└───cmd                   // Commands
+│   └───lukso             // LUKSO CLI
+│
+└───contracts             // Solidity Contracts
+│   └───bindings          // Bindings generated from ABIs
+|
+└───docs                  // Updates for Installation Progress
+|
+└───install               // Mandatory Installation Tools
+│   └───cf-wrangler       // Manager for Cloudflare Workers
+│   └───docs-processor    // Markdown to Page Converter
+│   └───macos_packages    // MacOS Codesigning Scripts
+|
+└───pid                   // Process ID Management
+```
+
+## CLI Installation Script
 
-## CLI Installation
 ```sh
-curl https://install.lukso.network | sh
+https://install.lukso.network
 ```
 
-## Downloading and Installing LUKSO
+**Running this script will install the full LUKSO CLI Tool on Mac and Linux.**
+**Installation directory: `/usr/local/bin/lukso`**
 
-```bash
-# 1. To install the LUKSO Command Line Interface (CLI)
-$ curl https://install.lukso.network | sh
+## Node Folder Structure
 
-# 2. Create a working folder where you want your clients to store their data
-$ mkdir myLUKSOFolder && cd ./myLUKSOFolder
+> Initializing a LUKSO node will not overwrite existing config, data or keystore folders
 
-# 3. This will initialize your working folder by downloading all network configs from
-# https://github.com/lukso-network/network-configs
-# NOTE: This will not overwrite any existing config, data or keystore folders
-$ lukso init
+```
+WIP
+```
 
-# 4. Install your clients. It will ask you which ones you want to install
-$ lukso install
+## External Sources
+
+- The network configuration is fetched from [lukso-network/network-configs](https://github.com/lukso-network/network-configs)
+
+- Deposit keys can be generated using [lukso-network/tools-key-gen-cli](https://github.com/lukso-network/tools-key-gen-cli)
+
+- Tool for generating validator keys [wagyu-key-gen](https://github.com/lukso-network/tools-wagyu-key-gen)
+
+## Client Clarification
+
+> WIP: More client setups will be added
+
+The LUKSO CLI is able to install multiple node clients.
+They cover the full node functionality of an EVM PoS Blockchain.
 
-# If you want to auto accept terms run it with
-$ lukso install --agree-terms
+- Supported Execution Node Clients: Geth
+- Supported Beacon Node Clients: Prysm
+- Validator Client for Staking
+
+## Setting up the Node
+
+Process of setting up the node using the LUSKO CLI Tool
+
+### Installing cURL
+
+Installing a tool to fetch the LUKSO CLI Installation Script from the server.
+
+#### MacOS
```

Comment:
not needed curls is installed on mac, probably same for linux.

the simpler the better

Reply:
Some linux versions prefer wget.
Internally the install.sh script actually uses wget for linux and curl for mac
```sh
download() {
  URL="$1";
  LOCATION="$2";
  if [[ $PLATFORM == "linux" ]]; then
    wget -O $LOCATION $URL;
  fi

  if [[ $PLATFORM == "darwin" ]]; then
    curl -o $LOCATION -Lk $URL;
  fi
}
```